### ISSUE TITLE
Progress bar is shown too late after layer was added

### ISSUE THREAD

**Issue by m-mohr** [jbccollins/leaflet-tile-loading-progress-control#6]
Sorry, I have to come up with an issue again. ;-)

I was wondering why the progress bar doesn't show up immediately for layers added later (see #4).

I would propose the following change in onAdd:

Replace 
```js
                map.on('layeradd', function(e) {
                    var currentLayers = leafletElt.getLayers();
                    for (var i = 0; i < currentLayers.length; i++) {
                        // If the layer added to the map is part of the layer group then we rebind the loading triggers.
                        if (currentLayers[i] === e.layer) {
                            controlInstance.unbindLoadEventTriggers();
                            controlInstance.bindLoadEventTriggers();
                            break;
                        }
                    }
                });
```

with

```js
                var self = this;
                map.on('layeradd', function(e) {
                    var currentLayers = leafletElt.getLayers();
                    for (var i = 0; i < currentLayers.length; i++) {
                        // If the layer added to the map is part of the layer group then we rebind the loading triggers.
                        if (currentLayers[i] === e.layer) {
                            controlInstance.unbindLoadEventTriggers();
                            controlInstance.bindLoadEventTriggers();
                            break;
                        }
                    }
                    self.handleLoadingStatusUpdate();
                });
```

It seems the first event fires quite late after the (slow loading) layer has been added. I could fix it by calling handleLoadingStatusUpdate() once after the layer has been added.

Still looking into the back-and-forth issue.

**Comment by jbccollins** [comment]
Seems like reasonable logic. But wouldn't it only be necessary to handle the loading status update inside of the conditional?
Like so:
```javascript
                map.on('layeradd', function(e) {
                    var currentLayers = leafletElt.getLayers();
                    for (var i = 0; i < currentLayers.length; i++) {
                        // If the layer added to the map is part of the layer group then we rebind the loading triggers.
                        if (currentLayers[i] === e.layer) {
                            controlInstance.unbindLoadEventTriggers();
                            controlInstance.bindLoadEventTriggers();
                            controlInstance.handleLoadingStatusUpdate();
                            break;
                        }
                    }
                });
```

**Comment by m-mohr** [comment]
Yes, indeed.

**Comment by jbccollins** [comment]
Fixed in 1.0.8

**Comment by m-mohr** [comment]
Great, thanks.
